hoprnet/hoprnet
# Node.js version checker

## Context

To keep the source code clean and easy to read, `hopr` makes use of new language features of Javascript as well as enhanced Node.js APIs.

When running with a wrong Node.js version, the process fails with an exception that does not directly reveal to the user that its root is an incompatible version of Node.js.

Example:
![image](https://user-images.githubusercontent.com/12514844/171035631-73d30839-91af-44a3-bc65-07ee9c980f16.png)

## Main change

This PR adds a small shim to the `hoprd` bootstrap sequence that checks the version of Node.js at runtime and fails with a comprehensible error if the version appears to be incompatible.

## How to test

```sh
fnm use # use correct version
yarn build # building requires modern Node.js
```
Check with multiple Node.js versions

Check various version of Node.js, e.g. Node.js 11 (without ESM support)

```sh
fnm use 11
node packages/hoprd/lib/main.cjs
```
